src/transformers/modeling_tf_utils.py

```diff
@@ -350,6 +353,44 @@ def __init__(self, config, *inputs, **kwargs):
         self.config = config
         self.name_or_path = config.name_or_path
 
+    def train_step(self, data):
+        """
+        Model training step.
+
+        Args:
+            data (:obj:`tuple`):
+                Tuple that contains the data to train on.
+        """
+        x, y = data
```

Comment:
data is always a tuple of size 3 maximum `(train_data, labels, sample_weights)` and `sample_weights` is added only if the parameter is filled in the `.fit()` method otherwise data is of size 2.

In our case, `train_data` are a dictionary and `labels` is either a Tensor or a dictionary. Dictionary in the case of question answering task for example.

For tasks such as question answering the `.compile()` method has to be called a bit differently because we have two outputs `start` and `end` logits. So the call becomes:

```
self.model.compile(optimizer=self.optimizer, loss={'output_1': self.model.compute_loss, 'output_2': self.model.compute_loss, metrics={"output_1": self.metrics, "output_2": self.metrics})
```

`output_1` and `output_2` are the name of the two outputs by default. But this will change to their real names `start_logits` and `end_logits`, I just need to do several changes for that.